public/app/features/inspector/InspectJSONTab.tsx

```diff
@@ -217,5 +217,18 @@ export class InspectJSONTab extends PureComponent<Props, State> {
 }
 
 function getPrettyJSON(obj: any): string {
-  return JSON.stringify(obj, null, 2);
+  let r = '';
+  try {
+    r = JSON.stringify(obj, null, 2);
+  } catch (e) {
```

Comment:
@TsotosA thanks for your contribution. I've tried this approach but gave up on it as it still crashed the browser tab when stringifying the object. And since the crash, I couldn't recover from it obviously.

Is your experience different? (sorry, haven't tested the PR yet).

Reply:
https://user-images.githubusercontent.com/22328404/193303661-61f10f88-29d8-4f14-8443-a02f1a51e971.mp4

hey @dprokop, i tried both Firefox and Chrome and got the same results, like the one in the video. It takes a few seconds trying to stringify and then recovers with the error message.